### PR TITLE
docs(revision control): remove revision control API, operator instructions

### PR DESCRIPTION
**Pull request by machikoyasuda**
Issue: #596 
Closes: #596
Connected with Reaction issue: https://github.com/reactioncommerce/reaction/pull/4238
To be released for 1.13 https://github.com/reactioncommerce/reaction/pull/4296

Remove revision control from:
- Table of contents
- Dev docs
- Operator docs
- Add deprecation warning to dev docs, operator docs
- Link to Changelog and PR that removed Revisions

### CODE REVIEW

File: admin/dashboard/catalog.md [hunk 1]
```diff
@@ -1,15 +1,3 @@
...
-At the moment revision control is merely used to stage a single upcoming update.
-
-## Enabling and disabling revision control
-
-In the dashboard click <i class="font-icon fa fa-book"></i> **Catalog**, where you may toggle revision control `on`, or `off`.
-
-Currently, turning off revision will give you the legacy version of the product detail page.
-
-![](/assets/admin-catalog-revision-settings.png "Catalog Revision Settings")
+> _Deprecation warning:_ Revision Control has been deprecated from Reaction as of 1.13. The legacy revision control system has been deprecated in favor of the new Catalog collection for overall product gird performance improvements and simplifed product publication logic. Read more about the changes in the CHANGELOG and [here](https://github.com/reactioncommerce/reaction/pull/4238).
```
Comment (mikemurray) [inline admin/dashboard/catalog.md]: The link [here] should say where "here" is. Maybe `[pull request]` or something else that makes sense?

Read more about the changes in the CHANGELOG and in this [pull request](https://github.com/reactioncommerce/reaction/pull/4238).

File: developer/revisions.md [hunk 2]
```diff
@@ -1,102 +1,3 @@
...
-### Check if revision control is enabled
-
-```js
-    import { isRevisionControlEnabled } from "/imports/plugins/core/revisions/lib/api";
-
-    if (isRevisionControlEnabled()) {
-      // Do something if revision control is enabled
-    }
-```
+> _Deprecation warning:_ Revision Control has been deprecated from Reaction as of 1.13. The legacy revision control system has been deprecated in favor of the new Catalog collection for overall product gird performance improvements and simplifed product publication logic. Read more about the changes in the CHANGELOG and [here](https://github.com/reactioncommerce/reaction/pull/4238).
```
Comment (mikemurray) [inline developer/revisions.md]: Same deal as above, `[here]` should say `[pull request]` or something to be clear where you're going.

File: admin/dashboard/catalog.md (hunk 1 above)
Comment (mikemurray) [reply admin/dashboard/catalog.md]: Also might as well link to the changelog too.

**Comment by machikoyasuda** [comment]
@mikemurray Thanks for the fast review! I've made the changes 🆙

**Pull request merged**